core/deployment/src/main/java/io/quarkus/deployment/util/WebJarUtil.java

```diff
@@ -0,0 +1,300 @@
+package io.quarkus.deployment.util;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileLock;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Scanner;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.bootstrap.model.AppArtifact;
+import io.quarkus.bootstrap.model.AppDependency;
+import io.quarkus.bootstrap.model.PathsCollection;
+import io.quarkus.bootstrap.util.IoUtils;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.pkg.builditem.CurateOutcomeBuildItem;
+import io.quarkus.runtime.LaunchMode;
+
+/**
+ * Utility for Web resource related operations
+ */
+public class WebJarUtil {
+
+    private static final Logger LOG = Logger.getLogger(WebJarUtil.class);
+
+    private final static String tmpDir = System.getProperty("java.io.tmpdir");
+    private static final String CUSTOM_MEDIA_FOLDER = "META-INF/branding/";
+    private static final List<String> IGNORE_LIST = Arrays.asList("logo.png", "favicon.ico", "style.css");
+
+    private WebJarUtil() {
+    }
+
+    public static Path devOrTest(CurateOutcomeBuildItem curateOutcomeBuildItem, LaunchModeBuildItem launch,
+            AppArtifact artifact, String rootFolderInJar)
+            throws IOException {
+
+        AppArtifact userApplication = curateOutcomeBuildItem.getEffectiveModel().getAppArtifact();
+
+        Path path = createDir(userApplication.getArtifactId(), artifact.getGroupId(), artifact.getArtifactId(),
+                artifact.getVersion());
+
+        // Clean on non dev mode
+        if (!launch.getLaunchMode().equals(LaunchMode.DEVELOPMENT)) {
+            IoUtils.recursiveDeleteAndThenCreate(path);
+        }
+
+        if (isEmpty(path)) {
+            ClassLoader classLoader = WebJarUtil.class.getClassLoader();
+            for (Path p : artifact.getPaths()) {
+                File artifactFile = p.toFile();
+                try (JarFile jarFile = new JarFile(artifactFile)) {
```

Comment:
Use `JarFiles.create` from Smallrye Common IO to support MR jars here

Reply:
Thanks ! Fixed.